docs/security.md

```diff
@@ -838,6 +838,17 @@ The following options provides finer-grained control for this feature:
   </td>
   <td>3.0.0</td>
 </tr>
+<tr>
+  <td><code>spark.kerberos.renewal.exclude.hadoopFileSystems</code></td>
+  <td>(none)</td>
+  <td>
+    A comma-separated list of Hadoop filesystems for whose hosts will be excluded from from delegation
+    token renewal at resource scheduler. For example, <code>spark.kerberos.renewal.exclude.hadoopFileSystems=hdfs://nn1.com:8032,
+    hdfs://nn2.com:8032</code>. This is known to work under YARN for now, so YARN Resource Manager won't renew tokens for the application.
+    Note that as resource scheduler does not renew token, the application might not be long running once the token expires.
```

Comment:
`might not be long running` is a little bit soft in my view. Can you imagine a situation where token expires and the workload goes on successfully?

Reply:
the workload can definitely go on until it tries to access that filesystem.  There is no reason it couldn't fully succeed depending on where its writing data or if it writes data at all.
We definitely could clarify though, perhaps something more like:
, so any application running longer than the original token expiration that tries to use that token will fail.

Reply:
so my intent was actually to add more description up on like line 792 and add to the "When using a Hadoop filesystem " bit.
Perhaps something like:

service hosting the user's home directory and staging directory.  Spark will renew Hadoop filesystem delegation tokens before their expiration unless the token is excluded via spark.kerberos.renewal.exclude.hadoopFileSystems. Please note that if the token is not renewed, any application that attempts to access the filesystem associated with that token after it expired will likely fail.

Reply:
As far as I understand, this config doesn't affect Spark's renewal of tokens, but avoid YARN RM's renewal. That is different. The token might be invalid if Spark doesn't renew it before expiration. Whether Spark renews the token is depending on https://github.com/apache/spark/pull/31761#discussion_r591350982.

Reply:
oh right.  Please update the description on the issue to be what you are really trying to do here.  

I'm not sure that is completely true though either.  getTokenRenewalInterval fetches the delegation tokens in fetchDelegationTokens and then calls renew on them to get the next expiration time. I'm assuming your changes in fetchDelegationTokens makes renew not work. What does renew return in this case that renewer is empty? It looks like it should throw an exception going by the base Hadoop classes I looked at.
Assuming its not the actual next renewal time since you say it doesn't get renewed if this tokens expiration is actually less than all the other tokens than we won't properly renew it on the Spark side either. That is in the case the key tab was specified.  If it throws an exception that is even worse.

I'm also wondering out loud if this config would ever apply to anything other than YARN. there is no renewal component in k8s that I'm aware of

Reply:
Okay, this sounds a good point to me. I think I should not change the behavior of `getTokenRenewalInterval` because it is not related to the issue here. The renew call in `getTokenRenewalInterval` is just used for obtaining renewal interval for all FS tokens.

I assume this config is only for YARN-specific behavior. I documented it explicitly about YARN in the config doc/security.md. Is any other thing I should do for it?

Reply:
I had a suggestion above for rephrasing the text here, something more like:

Note that as resource scheduler does not renew token, so any application running longer than the original token expiration that tries to use that token will likely fail.